src/go/rpk/pkg/cli/cloud/login.go

```diff
@@ -65,153 +65,90 @@ If none of these are provided, rpk will use the SSO method to login.
-want to disable automatic profile creation and selection, use --no-profile.
 `,
 		Run: func(cmd *cobra.Command, _ []string) {
 			cfg, err := p.Load(fs)
-			out.MaybeDie(err, "unable to load config: %v", err)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
 			y, err := cfg.ActualRpkYamlOrEmpty()
-			out.MaybeDie(err, "unable to load config: %v", err)
-			auth := y.Auth(y.CurrentCloudAuth)
-			var cc bool
-			if auth != nil {
-				cc = auth.HasClientCredentials()
-			}
-			_, err = oauth.LoadFlow(cmd.Context(), fs, cfg, auth0.NewClient(cfg.DevOverrides()), noBrowser)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+
+			p := y.Profile(y.CurrentProfile)
+			authAct, authVir, clearedProfile, isNewAuth, err := oauth.LoadFlow(cmd.Context(), fs, cfg, auth0.NewClient(cfg.DevOverrides()), noBrowser, forceReload, cfg.DevOverrides().CloudAPIURL)
 			if err != nil {
 				fmt.Printf("Unable to login to Redpanda Cloud (%v).\n", err)
-				if e := (*oauth.BadClientTokenError)(nil); errors.As(err, &e) && cc {
+				if e := (*oauth.BadClientTokenError)(nil); errors.As(err, &e) && authVir != nil && authVir.HasClientCredentials() {
 					fmt.Println(`You may need to clear your client ID and secret with 'rpk cloud logout --clear-credentials',
 and then re-specify the client credentials next time you log in.`)
 				} else {
 					fmt.Println(`You may need to clear your credentials with 'rpk cloud logout --clear-credentials', and login again`)
 				}
 				os.Exit(1)
 			}
-
-			if cc && save {
-				yAct, _ := cfg.ActualRpkYaml() // must exist due to LoadFlow checking
-				yAct.Auth(yAct.CurrentCloudAuth).ClientSecret = auth.ClientSecret
-				err = yAct.Write(fs)
+			if authVir.HasClientCredentials() && save {
+				authAct.ClientSecret = authVir.ClientSecret
+				err = y.Write(fs)
 				out.MaybeDie(err, "unable to save client ID and client secret: %v", err)
 			}
-			fmt.Println("Successfully logged in.")
-			if noProfile {
+
+			fmt.Printf("Successfully logged into organization %q (%s) via %s.\n", authAct.Organization, authAct.OrgID, authAct.AnyKind())
+			if !isNewAuth {
+				fmt.Println("If you are trying to authenticate to a different organization, use 'rpk cloud login --reload'.")
+			}
+			fmt.Println()
+
+			// When you have no profile, or you have one that is not selected.
+			if p == nil || len(y.Profiles) == 0 {
+				fmt.Println(`To create an rpk profile to talk to an existing cloud cluster, use 'rpk cloud cluster use'.
+To learn more about profiles, check 'rpk profile --help'.
+You are not currently in a profile; rpk talks to a localhost:9092 cluster by default.`)
+				return
+			}
+
+			// When you had a profile, but it was cleared due to your browser
+			// having a different org's auth.
+			if p != nil && clearedProfile {
+				priorAuth := p.ActualAuth()
+				fmt.Printf(`rpk swapped away from your prior profile %q which authenticated with organization %q (%s).
```

Comment:
Reduce verbosity, more directive language:

```
Login token change detected

Select cloud cluster to talk to:
    rpk cloud cluster use

rpk will talk to a localhost:9092 cluster otherwise

```

Reply:
Actually this one, I'd like to keep the current messaging (or the details of it) -- a lot of the confusion people were running into was what happened when they switched orgs, and I want to make sure it's really clear to end users that their auth is now talking to a different org.